### PR TITLE
Update arwen, fix upgrade protection

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Arwen [v0.3.29](https://github.com/ElrondNetwork/arwen-wasm-vm/releases/tag/v0.3.29)
 - Arwen [v0.3.30](https://github.com/ElrondNetwork/arwen-wasm-vm/releases/tag/v0.3.30)
 - Enforced SC upgrade protection in Smart Contract Processor - protect against storage updates as well (not only for code updates)
 - Added extra tests
 - Simplify long running test `TestSCCallingDNSUserNames`

### CODE REVIEW

File: integrationTests/oneNodeNetwork.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+func (network *oneNodeNetwork) Mint(address []byte, value *big.Int) {
+	MintAddress(network.Node.AccntState, address, value)
+}
+
+func (network *oneNodeNetwork) GetMinGasPrice() uint64 {
+	return network.Node.EconomicsData.GetMinGasPrice()
+}
+
+func (network *oneNodeNetwork) MaxGasLimitPerBlock() uint64 {
```
Comment (sasurobert) [inline integrationTests/oneNodeNetwork.go:73]: n *oneNodeNetwork. add comment to all public functions

File: process/smartContract/errors.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package smartContract
+
+import (
+	"errors"
+)
+
+// ErrUpgradeNotAllowed signals that upgrade is not allowed
+var ErrUpgradeNotAllowed = errors.New("upgrade not allowed")
```
Comment (sasurobert) [inline process/smartContract/errors.go:8]: more this error to process/errors.go

Comment (sasurobert) [reply process/smartContract/errors.go:8]: upgrade is allowed only by user ?

File: process/smartContract/process.go [hunk 3]
```diff
@@ -230,9 +233,21 @@ func (sc *scProcessor) doExecuteSmartContractTransaction(
...
+	}
+
 	executedBuiltIn, err = sc.resolveBuiltInFunctions(txHash, tx, acntSnd, acntDst, vmInput)
 	if err != nil {
-		log.Debug("processed built in functions error", "error", err.Error())
+		// QUESTION FOR REVIEW: before, this was log.Debug() and, in our tests, resulted in a lot of:
+		// "processed built in functions error error = insufficient funds",
+		// Is it OK to have this new condition on log.Debug() here?
+		if !executedBuiltIn {
+			log.Debug("processed built in functions error", "error", err.Error())
```
Comment (sasurobert) [inline process/smartContract/process.go:248]: move to log.Trace - no need for log .debug

File: integrationTests/oneNodeNetwork.go (hunk 1 above)
Comment (andreibancioiu) [reply integrationTests/oneNodeNetwork.go:73]: Fixed.

File: process/smartContract/errors.go (hunk 2 above)
Comment (andreibancioiu) [reply process/smartContract/errors.go:8]: Fixed to "upgrade is allowed only for owner".

File: process/smartContract/process.go (hunk 3 above)
Comment (andreibancioiu) [reply process/smartContract/process.go:248]: Fixed.

**Pull request merged**